cli/lsp/config.rs

```diff
@@ -1083,512 +1094,468 @@ impl LspTsConfig {
   }
 }
 
-#[derive(Debug, Clone)]
-pub struct LspPackageConfig {
-  pub nv: PackageNv,
-  pub exports: Value,
-}
-
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub enum ConfigWatchedFileType {
   DenoJson,
   Lockfile,
+  NpmRc,
   PackageJson,
   ImportMap,
 }
 
 /// Contains the config file and dependent information.
 #[derive(Debug, Clone)]
 pub struct ConfigData {
-  pub scope: ModuleSpecifier,
-  pub config_file: Option<Arc<ConfigFile>>,
-  pub fmt_options: Arc<FmtOptions>,
-  pub lint_options: Arc<LintOptions>,
-  pub lint_config: LintConfig,
+  pub scope: Arc<ModuleSpecifier>,
+  pub workspace: Arc<Workspace>,
+  pub member_ctx: Arc<WorkspaceMemberContext>,
+  pub fmt_config: Arc<deno_config::FmtConfig>,
+  pub lint_config: Arc<deno_config::LintConfig>,
+  pub test_config: Arc<deno_config::TestConfig>,
+  pub exclude_files: Arc<PathOrPatternSet>,
+  pub deno_lint_config: deno_lint::linter::LintConfig,
   pub lint_rules: Arc<ConfiguredRules>,
   pub ts_config: Arc<LspTsConfig>,
   pub byonm: bool,
   pub node_modules_dir: Option<PathBuf>,
   pub vendor_dir: Option<PathBuf>,
   pub lockfile: Option<Arc<CliLockfile>>,
-  pub package_json: Option<Arc<PackageJson>>,
   pub npmrc: Option<Arc<ResolvedNpmRc>>,
-  pub import_map: Option<Arc<ImportMap>>,
-  pub import_map_from_settings: bool,
-  pub package_config: Option<Arc<LspPackageConfig>>,
-  pub is_workspace_root: bool,
-  pub workspace_root_dir: ModuleSpecifier,
-  /// Workspace member directories. For a workspace root this will be a list of
-  /// members. For a member this will be the same list, representing self and
-  /// siblings. For a solitary package this will be `vec![self.scope]`. These
-  /// are the list of packages to override with local resolutions for this
-  /// config scope.
-  pub workspace_members: Arc<Vec<ModuleSpecifier>>,
+  pub resolver: Arc<WorkspaceResolver>,
+  pub import_map_from_settings: Option<ModuleSpecifier>,
   watched_files: HashMap<ModuleSpecifier, ConfigWatchedFileType>,
 }
 
 impl ConfigData {
   async fn load(
-    config_file_specifier: Option<&ModuleSpecifier>,
+    specified_config: Option<&Path>,
     scope: &ModuleSpecifier,
-    workspace_root: Option<&ConfigData>,
     settings: &Settings,
-    file_fetcher: Option<&Arc<FileFetcher>>,
+    file_fetcher: &Arc<FileFetcher>,
+    // sync requirement is because the lsp requires sync
+    cached_deno_config_fs: &(dyn DenoConfigFs + Sync),
+    deno_json_cache: &(dyn DenoJsonCache + Sync),
+    pkg_json_cache: &(dyn PackageJsonCache + Sync),
```

Comment:
Sidenote: maybe we could create a wrapper like `deno_unsync` that would cheat the type system and not require these `Sync` trait bounds?